record/record.go

```diff
@@ -198,9 +206,27 @@ func (r *Reader) Next() (io.Reader, error) {
 		return nil, r.err
 	}
 	r.started = true
+	r.inRecovery = false
 	return singleReader{r, r.seq}, nil
 }
 
+// Recover clears any errors read so far, so that calling Next will start
+// reading from the next good 32KiB block. If there are no such blocks, Next
+// will return io.EOF.
+//
+// Recover also marks the current reader, the one most recently returned by
+// Next, as stale.
+func (r *Reader) Recover() {
+	// After calling r.Recover, we're effectively starting fresh, so we need to
```

Comment:
I'd replace these three lines of comments with just:

r.err = nil
r.inRecovery = true
// Discard the rest of the current block.
r.i, r.j, r.n, r.last = 0, 0, 0, false
// Invalidate any outstanding singleReader.
r.seq++

Or perhaps that r.i etc line should be:

r.i, r.j, r.last = r.n, r.n, false

and you might not need to touch r.inRecovery or r.started at all.